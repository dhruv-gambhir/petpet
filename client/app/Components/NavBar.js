"use client";

import Title from "./Title";
import { useRouter } from "next/navigation";

export default function NavBar() {
    const router = useRouter();

    return (
        <nav className="bg-white w-full h-40 flex-none flex flex-row items-center justify-center">
            <div className="flex flex-row justify-between w-full">
                <Title className="w-1/2 left-2" />
                <div className="w-1/2 h-full right-2 flex flex-row">
                    <button
                        className="h-16 w-1/6 bg-gray-300 border-1 border-black rounded mx-8 justify-end"
                        onClick={() => {
                            router.push("/sitters/owner");
                        }}
                    >
                        Sitters
                    </button>
                    <button
                        className="w-1/6 bg-gray-300 border-1 border-black rounded mx-8"
                        onClick={() => {
                            router.push("/events");
                        }}
                    >
                        Events
                    </button>
                    <button
                        className="w-1/6 bg-gray-300 border-1 border-black rounded mx-8"
                        onClick={() => {
                            router.push("/adoption");
                        }}
                    >
                        Adoption
                    </button>
                    <button
                        className="w-1/6 bg-gray-300 border-1 border-black rounded mx-8"
                        onClick={() => {
                            router.push("/profile");
                        }}
                    >
                        Profile
                    </button>
                </div>
            </div>
        </nav>
    );
}
