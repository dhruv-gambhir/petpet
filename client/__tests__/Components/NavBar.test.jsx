// __tests__/components/NavBar.test.jsx

import React from 'react';
import { render, screen } from '@testing-library/react';
import NavBar from '../../app/Components/NavBar';
import useStore from '../../app/store';

// Mock the useStore hook
jest.mock('../../app/store');

// Mock the Title component
jest.mock('../../app/Components/Title', () => {
  return function MockTitle(props) {
    return <div data-testid="mock-title" {...props}>Mocked Title</div>;
  };
});

// Mock Next.js Link component
jest.mock('next/link', () => {
  return function MockLink({ children, href }) {
    return <a href={href}>{children}</a>;
  };
});

describe('NavBar Component', () => {
  // Set up the mocked store before each test
  beforeEach(() => {
    useStore.mockReturnValue({ zId: 'user123' });
  });

  // Clear all mocks after each test to prevent interference
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders NavBar without crashing', () => {
    render(<NavBar />);
    const navElement = screen.getByRole('navigation');
    expect(navElement).toBeInTheDocument();
  });

  test('renders the Title component', () => {
    render(<NavBar />);
    const titleElement = screen.getByTestId('mock-title');
    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveTextContent('Mocked Title');
  });

  test('renders all navigation links', () => {
    render(<NavBar />);
    const sittersLink = screen.getByText('Sitters');
    const eventsLink = screen.getByText('Events');
    const adoptionLink = screen.getByText('Adoption');
    const profileLink = screen.getByText('Profile');

    expect(sittersLink).toBeInTheDocument();
    expect(eventsLink).toBeInTheDocument();
    expect(adoptionLink).toBeInTheDocument();
    expect(profileLink).toBeInTheDocument();
  });

  test('navigation links have correct href attributes', () => {
    render(<NavBar />);
    const sittersLink = screen.getByText('Sitters').closest('a');
    const eventsLink = screen.getByText('Events').closest('a');
    const adoptionLink = screen.getByText('Adoption').closest('a');
    const profileLink = screen.getByText('Profile').closest('a');

    expect(sittersLink).toHaveAttribute('href', '/sitters/owner');
    expect(eventsLink).toHaveAttribute('href', '/events');
    expect(adoptionLink).toHaveAttribute('href', '/adoption');
    expect(profileLink).toHaveAttribute('href', '/profile/user123');
  });

  test('Profile link includes zId from the store', () => {
    render(<NavBar />);
    const profileLink = screen.getByText('Profile').closest('a');
    expect(profileLink).toHaveAttribute('href', '/profile/user123');
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<NavBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});

