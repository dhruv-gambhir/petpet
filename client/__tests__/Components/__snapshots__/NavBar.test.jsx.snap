// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="bg-mynav w-full h-40 flex-none flex flex-row items-center justify-center"
  >
    <div
      class="flex flex-row justify-between w-full"
    >
      <div
        class="w-1/2 left-2"
        data-testid="mock-title"
      >
        Mocked Title
      </div>
      <div
        class="w-1/2 h-full right-2 flex flex-row"
      >
        <a
          href="/sitters/owner"
        >
          Sitters
        </a>
        <a
          href="/events"
        >
          Events
        </a>
        <a
          href="/adoption"
        >
          Adoption
        </a>
        <a
          href="/profile/user123"
        >
          Profile
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
